### ISSUE TITLE
Invoking Lambda function does not generate the correct payload

### ISSUE THREAD

**Issue by RealSalmon** [spulec/moto#769]
Is the payload delivered when invoking a lambda function incorrect? Instead of delivering the function's return value, it looks like it is delivering a dictionary with a single 'result' key that contains the log output.

e.g. I would expect test_invoke() below to pass . . .

```python

import io
import zipfile
import boto3
import json
from moto import mock_lambda

lambda_fx = """
def lambda_handler(event, context):
    return event
"""

def init_lambda(fx):
    zip_output = io.BytesIO()
    zip_file = zipfile.ZipFile(zip_output, 'w', zipfile.ZIP_DEFLATED)
    zip_file.writestr('lambda_function.zip', fx)
    zip_file.close()
    zip_output.seek(0)

    boto3.client('lambda').create_function(
        FunctionName='test-lambda-fx',
        Runtime='python2.7',
        Role='test-iam-role',
        Handler='lambda_function.lambda_handler',
        Description='test lambda function',
        Timeout=3,
        MemorySize=128,
        Publish=True,
        Code={
            'ZipFile': zip_output.read()
        },
    )


@mock_lambda
def test_invoke():
    init_lambda(lambda_fx)
    result = boto3.client('lambda').invoke(
        FunctionName='test-lambda-fx',
        InvocationType='RequestResponse',
        LogType='Tail',
        Payload=json.dumps({'hello': 'world'})
    )

    assert json.loads(result['Payload'].read()) == {'hello': 'world'}

```

**Comment by rocky4570** [comment]
Ah ok - the patch I put up was putting data to LogResult (http://boto3.readthedocs.io/en/latest/reference/services/lambda.html#Lambda.Client.invoke) so I will have a look at this.

**Comment by spulec** [comment]
Fixed with #781
